src/pudl_archiver/archivers/sec10k.py

```diff
@@ -0,0 +1,50 @@
+"""Download SEC10k extracted tables for arhival."""
+
+import deltalake
+import pandas as pd
+
+from pudl_archiver.archivers.classes import (
+    AbstractDatasetArchiver,
+    ArchiveAwaitable,
+    ResourceInfo,
+)
+
+# This can be removed once we change the upstream table names to reflect their raw status
+TABLE_NAME_MAP = {
+    "core_sec10k__filings": "raw_sec10k__quarterly_filings",
+    "core_sec10k__company_information": "raw_sec10k__quarterly_company_information",
+    "out_sec10k__parents_and_subsidiaries": "raw_sec10k__parents_and_subsidiaries",
+    "core_sec10k__exhibit_21_company_ownership": "raw_sec10k__exhibit_21_company_ownership",
+}
+
+
+def _year_quarters_from_dataframe(df: pd.DataFrame) -> dict:
+    return {"year_quarters": set(df["year_quarter"].unique())}
+
+
+class Sec10kArchiver(AbstractDatasetArchiver):
+    """Sec10k raw extracted archiver."""
+
+    name = "sec10k"
+    deltalake_version = 0
+
+    async def get_resources(self) -> ArchiveAwaitable:
+        """Archive monolithic parquet files for each raw SEC 10k table."""
+        for delta_name, raw_name in TABLE_NAME_MAP.items():
+            yield self.get_delta_table(delta_name, raw_name)
+
+    async def get_delta_table(self, delta_name: str, raw_name: str):
+        """Read configured version of table from deltalake on GCS and save parquet."""
+        table_url = f"gs://model-outputs.catalyst.coop/sec10k/{delta_name}"
+        download_path = self.download_directory / f"{raw_name}.parquet"
+        dt = deltalake.DeltaTable(table_url, version=self.deltalake_version)
+        df = dt.to_pandas()
+        df.to_parquet(download_path)
+        date_parts = {}
+        if delta_name != "out_sec10k__parents_and_subsidiaries":
+            date_parts = _year_quarters_from_dataframe(df)
+
+        return ResourceInfo(
+            local_path=download_path,
+            partitions={"table_name": raw_name} | date_parts,
```

Comment:
Do we really want to partition by table? Most of our data sources have many tables contained within a given partition -- whether they're stored in different spreadsheets, FERC database tables, etc.  Why not just partition by date and have each of the 4 tables available within each year partition?

The `out_sec10k__parents_and_subsidiaries` table contains a `report_date` column. Should that not be the basis for partitioning that table? Otherwise do you just always get all of that table?

Reply:
If we include partitions here that don't mean anything, I think the datastore infrastructure won't work correctly.  Looking at the FERC-714 bulk-CSV archiver (for 2006-2020 data) it sets the partitions to an empty dictionary.

Reply:
My thinking here was that in most cases when we have many tables within a single partition, the raw data is zipped together into a single '`Resource`', whereas here each table has it's own parquet file and is it's own `Resource`.

I guess if we wanted to make this more consistent with other datasets and have all the tables bundled in one `Resource` we could either zip the parquet files together or load the data into a duckdb file.

Reply:
Okay, I can see the per-file (table) partitions being convenient in this case even if it's different from almost all of our other archivers.  But what about the date component?

Reply:
Fixed so every table now has `years` partitions

Reply:
I though we were trying to get rid of the date partitions entirely and treat it monolithically like the older FERC-714 data. if there are date partitions here then the datastore will expect to be able to pull them individually but they won't exist.

Reply:
Oh right. I had made it so there was a list of all years associated with each table, but that's probably unnecessarily complex and confusing, so I've removed them entirely.

Reply:
Okay great -- so it'll be partitioned by table, and we'll have years (or quarters) separately defined in the ETL Settings object, which can be used to select only a subset of the data for processing in CI, etc.

Reply:
I think you need to make the analogous changes to the data source in PUDL for this to work though.

Reply:
Also if the cached environment is screwing things up I think you can temporarily tell it not to do that in the GHA workflow file, and then it'll build it from scratch?